src/daemon/ca/events.rs

```diff
@@ -71,19 +74,195 @@ impl fmt::Display for IniDet {
 /// Describes an update to the set of ROAs under a ResourceClass.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct RoaUpdates {
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "updated_map"
+    )]
     updated: HashMap<RouteAuthorization, RoaInfo>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "removed_map"
+    )]
     removed: HashMap<RouteAuthorization, RevokedObject>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "aggregate_updated_map"
+    )]
     aggregate_updated: HashMap<RoaAggregateKey, AggregateRoaInfo>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "aggregate_removed_map"
+    )]
     aggregate_removed: HashMap<RoaAggregateKey, RevokedObject>,
 }
 
+mod updated_map {
```

Comment:
If sorting is a deliberate property of these serializers, would it perhaps be an idea to include `sorted` in the module name so that this property is immediately apparent when reading the serde `with` attributes above?